src/plopp/core/node.py

```diff
@@ -105,17 +105,22 @@ def remove(self) -> None:
         The operation fails is the node has children, as removing it would leave the
         graph in an ill-defined state.
         """
-        if self.children:
-            raise RuntimeError(
-                f"Cannot delete node because it has children {self.children}."
-            )
+        for child in self.children:
+            if self in child.parents:
```

Comment:
Why is this check needed?

Reply:
Because we don't know if `self` is in the `parents` or the `kwparents`.

Reply:
Ah..! right. You handle the `kwparents` below. Thank you!